### ISSUE TITLE
xor result is incorrect

### ISSUE THREAD

**Issue by cyliu0** [pingcap/tidb#13193]
## Bug Report

Please answer these questions before submitting your issue. Thanks!
There is a related PR https://github.com/pingcap/tidb/pull/12173

1. What did you do?
If possible, provide a recipe for reproducing the error.
```
mysql> select "0.000001" xor 0;
```

2. What did you expect to see?
```
mysql> select "0.000001" xor 0;
+------------------+
| "0.000001" xor 0 |
+------------------+
|                1 |
+------------------+
1 row in set (0.00 sec)
```

3. What did you see instead?
```
mysql> select "0.000001" xor 0;
+------------------+
| "0.000001" xor 0 |
+------------------+
|                0 |
+------------------+
1 row in set, 1 warning (0.00 sec)
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
```
mysql> select tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v4.0.0-alpha-738-g9cc0f6458
Git Commit Hash: 9cc0f645820ec2acf7a982acc04cbf95ea8da3c9
Git Branch: master
UTC Build Time: 2019-11-06 08:09:01
GoVersion: go version go1.13.1 darwin/amd64
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
1 row in set (0.00 sec)
```

**Comment by SunRunAway** [comment]
Duplicate of #12867

**Comment by SunRunAway** [comment]
In short, it's on purpose and compatible with mysql5.7. See #12867

**Comment by SunRunAway** [comment]
We've decided to follow the bug fix in MySQL8, #12867 is reopened.
